cfme/tests/intelligence/reports/test_metering_report.py

```diff
@@ -0,0 +1,314 @@
+        provider):
+    yield
+    BaseProvider.clear_providers()
+
+
+def new_credential():
+    return Credential(principal='uid' + '{}'.format(fauxfactory.gen_alphanumeric()),
+        secret='secret')
+
+
+@pytest.yield_fixture(scope="module")
+def vm_ownership(enable_candu, clean_setup_provider, provider, appliance):
+    # In these tests, chargeback reports are filtered on VM owner.So,VMs have to be
+    # assigned ownership.
+
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    if not provider.mgmt.does_vm_exist(vm_name):
+        pytest.skip("Skipping test, cu-24x7 VM does not exist")
+    if not provider.mgmt.is_vm_running(vm_name):
+        provider.mgmt.start_vm(vm_name)
+        provider.mgmt.wait_vm_running(vm_name)
+
+    group_collection = appliance.collections.groups
+    cb_group = group_collection.instantiate(description='EvmGroup-user')
+    user = appliance.collections.users.create(
+        name=fauxfactory.gen_alphanumeric(),
+        credential=new_credential(),
+        email='abc@example.com',
+        group=cb_group,
+        cost_center='Workload',
+        value_assign='Database')
+
+    vm = VM.factory(vm_name, provider)
+
+    try:
+        vm.set_ownership(user=user.name)
+        logger.info('Assigned VM OWNERSHIP for {} running on {}'.format(vm_name, provider.name))
+
+        yield user.name
+    finally:
+        vm.unset_ownership()
+        user.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def enable_candu(provider, appliance):
+    # C&U data collection consumes a lot of memory and CPU.So, we are disabling some server roles
+    # that are not needed for Chargeback reporting.
+
+    server_info = appliance.server.settings
+    original_roles = server_info.server_roles_db
+    server_info.enable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector', 'ems_metrics_processor')
+    server_info.disable_server_roles('automate', 'smartstate')
+    command = ('Metric::Targets.perf_capture_always = {:storage=>true, :host_and_cluster=>true};')
+    appliance.ssh_client.run_rails_command(command, timeout=None)
+
+    yield
+
+    server_info.update_server_roles_db(original_roles)
+    command = ('Metric::Targets.perf_capture_always = {:storage=>false, :host_and_cluster=>false};')
+    appliance.ssh_client.run_rails_command(command, timeout=None)
+
+
+def verify_records_rollups_table(appliance, provider):
+    # Verify that hourly rollups are present in the metric_rollups table.
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    ems = appliance.db.client['ext_management_systems']
+    rollups = appliance.db.client['metric_rollups']
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vm_name,
```

Comment:
**Required-ish**

Split these filter args onto multiple lines for readability, like:
```python
result = (
    call()
    .filter(thing.field == check,
             thing.field2 == check2,
             thing.field3 == check3)
)

Reply:
Done